### PR TITLE
Refactor DetCon logic & Update SupCon for seg recipes

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes
- enable to use two views per image instead of single view
- update lr warmup iters in supcon recipe
- create `DetConHead` including detcon loss and predictor
- update configs & docstrings
- add unit test

Through these changes, we can expect improved mDice (~+4.96) and reduced training time (~0.68x) compared with base supervised learning.

### How to test

`tests/unit/algorithms/segmentation/adapters/mmseg/models/heads/test_detcon_head.py`
image

`tests/unit/algorithms/segmentation/adapters/mmseg/models/segmentors/test_detcon.py`
image

`tests/unit/algorithms/segmentation/adapters/mmseg/models/losses/test_detcon_loss.py`
image

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [x] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by sungmanc** [comment]
I changed the title of this PR, we're not using [FIX], [ENHANCE] prefix anymore.

**Review by sungmanc — DISMISSED** [review]
LGTM, it showed quite improvement for SupCon algorithm. How about adding this changes to CHANGELOG ?

**Review by jaegukhyun — DISMISSED** [review]

**Comment by jaegukhyun** [comment]
Ah, there is test failure, could you check it?

**Comment by sungchul1** [comment]
@sungmanc @jaegukhyun 
I updated CHANGELOG and self-sl recipes to resolve integration test errors.
Could you review it again?

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ All notable changes to this project will be documented in this file.
 ### Enhancements
 
 - Clean up and refactor the output of the OTX CLI (<https://github.com/openvinotoolkit/training_extensions/pull/1946>)
+- Update DetCon logic and SupCon for semantic segmentation recipes (<https://github.com/openvinotoolkit/training_extensions/pull/1958>)
```
Comment (sungmanc) [inline CHANGELOG.md]:
```suggestion
- Enhance DetCon logic and SupCon for semantic segmentation(<https://github.com/openvinotoolkit/training_extensions/pull/1958>)
```
To emphasize the `improvement`, I suggested this

**Review by sungmanc — APPROVED** [review]

**Review by supersoob — APPROVED** [review]
LGTM

**Review by harimkang — APPROVED** [review]

**Pull request merged**